client/html/src/Client/Html/Basket/Mini/Product/Default.php

```diff
@@ -0,0 +1,192 @@
+	 */
+	public function getHeader($uid = '', array &$tags = array(), &$expire = null)
+	{
+		$view = $this->_setViewParams($this->getView(), $tags, $expire);
+
+		$html = '';
+		foreach ($this->_getSubClients() as $subclient) {
+			$html .= $subclient->setView($view)->getHeader($uid, $tags, $expire);
+		}
+		$view->mainHeader = $html;
+
+		/** client/html/basket/mini/product/default/template-header
+		 * Relative path to the HTML header template of the basket mini product client.
+		 *
+		 * The template file contains the HTML code and processing instructions
+		 * to generate the HTML code that is inserted into the HTML page header
+		 * of the rendered page in the frontend. The configuration string is the
+		 * path to the template file relative to the layouts directory (usually
+		 * in client/html/layouts).
+		 *
+		 * You can overwrite the template file configuration in extensions and
+		 * provide alternative templates. These alternative templates should be
+		 * named like the default one but with the string "default" replaced by
+		 * an unique name. You may use the name of your project for this. If
+		 * you've implemented an alternative client class as well, "default"
+		 * should be replaced by the name of the new class.
+		 *
+		 * @param string Relative path to the template creating code for the HTML page head
+		 * @since 2014.03
+		 * @category Developer
+		 * @see client/html/basket/mini/product/default/template-body
+		 */
+		$tplconf = 'client/html/basket/mini/product/default/template-header';
+		$default = 'basket/mini/product-header-default.html';
+
+		return $view->render($this->_getTemplate($tplconf, $default));
+	}
+
+
+	/**
+	 * Returns the sub-client given by its name.
+	 *
+	 * @param string $type Name of the client type
+	 * @param string|null $name Name of the sub-client (Default if null)
+	 * @return Client_Html_Interface Sub-client object
+	 */
+	public function getSubClient($type, $name = null)
+	{
+		return $this->_createSubClient('basket/mini/product/' . $type, $name);
+	}
+
+
+	/**
+	 * Returns the list of sub-client names configured for the client.
+	 *
+	 * @return array List of HTML client names
+	 */
+	protected function _getSubClientNames()
+	{
+		return $this->_getContext()->getConfig()->get($this->_subPartPath, $this->_subPartNames);
+	}
+
+
+	/**
+	 * Sets the necessary parameter values in the view.
+	 *
+	 * @param MW_View_Interface $view The view object which generates the HTML output
+	 * @param array &$tags Result array for the list of tags that are associated to the output
+	 * @param string|null &$expire Result variable for the expiration date of the output (null for no expiry)
+	 * @return MW_View_Interface Modified view object
+	 */
+	protected function _setViewParams(MW_View_Interface $view, array &$tags = array(), &$expire = null)
+	{
+		if (!isset($this->_cache)) {
+			$view->mainPriceItem = $view->miniBasket->getPrice();
```

Comment:
Uhh, I wear glasses so that's an extreme exaggeration ;-)
You can use productItems. productProducts would be in line with the rest but sounds strange ...